public/index.php

```diff
@@ -51,6 +53,22 @@
 
 $kernel = new Kernel($_SERVER['APP_ENV'], (bool)$_SERVER['APP_DEBUG']);
 $request = Request::createFromGlobals();
+
+$isBlackfireProfilingRequested = $request->headers->has('X-Blackfire');
+
+if ($isBlackfireProfilingRequested) {
```

Comment:
Suggestion: as the final user can manipulate the headers I'd suggest to have also an env variable o enable it Ex:

if ($isBlackfireProfilingRequested && $_ENV['BLACKFIRE_ENABLED']) {
...
}

Reply:
Good suggestion, fixed!